common/test/mock-ingester/src/lib.rs

```diff
@@ -25,7 +25,9 @@ use tokio_stream::wrappers::TcpListenerStream;
 const ROOT: &str = "/logs/agent";
 
 pub type FileLineCounter = Arc<Mutex<HashMap<String, FileInfo>>>;
-pub type ProcessFn = Box<dyn Fn(&IngestBody) + Send + Sync>;
+pub type ProcessFn = Box<
+    dyn (Fn(&IngestBody) -> Option<Pin<Box<dyn Future<Output = ()> + Send + Sync>>>) + Send + Sync,
```

Comment:
Allow the process fn to optionally return a future to be awaited

Reply:
Would it make sense to define the return type separately?
```
pub type ProcessRet = Option<Pin<Box<dyn Future<Output = ()> + Send + Sync>>>;
pub type ProcessFn = Box <dyn (Fn(&IngestBody) -> ProcessRet) + Send + Sync>;
```